### ISSUE TITLE
Sort out "should hex strings have 0x prefixes"

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#2498]
When doing this issue we need to consider at least:
- Types that wrap stdlib integer types eg, `Sequence`, and lock time types.
- Hex strings representing fixed size arrays eg, `PublicKey`

Also, we should consider where the solution lives, here in `rust-bitcoin`, in `units`, or in `hex-conservative`.

This issue will involve complex error stuff because we will likely want to carry the input string in the error which is a PITA with no-std.

**Comment by Kixunil** [comment]
We could also just (temporarily) give up on providing hex for integers. Arrays are definitely without prefix because it's never used anywhere in practice.

> This issue will involve complex error stuff because we will likely want to carry the input string in the error which is a PITA with no-std.

It is not but I think we will have to stabilize `InputString` (which is fine IMO, it's a trivial type, it just needs an independent crate).

**Comment by tcharding** [comment]
Consensus in #2492 is to have inherent methods on types that wrap integer types: 

- `from_hex` - must contain prefix
- `from_unprefixed_hex` - must not contain prefix

**Comment by sanket1729** [comment]
I have a slight prefix otherways. `from_hex_with_prefix` and `from_hex`. But I don't feel strongly about it to oppose the current consensus.

**Comment by mpbagot** [comment]
I think this has been closed by #5782

**Comment by tcharding** [comment]
I'm happy to close, leaving for @apoelstra to think for 30 seconds too.

**Comment by apoelstra** [comment]
Yeah, phew, I saw this in my notifications and thought "wtf we just resolved this".
